### ISSUE TITLE
Multiple chat histories, support gpt-3.5-turbo

### ISSUE THREAD

**Issue by bytesuji** [enricoros/nextjs-chatgpt-app#2]
Would be nice to have the chat history sidebar like ChatGPT. Also would like to be able to use gpt-3.5-turbo occasionally, since it's faster.

**Comment by enricoros** [comment]
Those are two great ideas. I'll merge the pull request in a couple of hours, great point about the speed of the model.

I find myself using the full context window of GPT-4 frequently, it's so good at chewing large amounts of data!

Any idea on how to get the conversation sidebar? UX should be doable / easy, but I wonder where to store the data (there's no central storages/db right now)

**Comment by peperunas** [comment]
I think the lack of an external DB or storage is one of the selling points of the project.

It allows more privacy for the users' prompts that are, in this way, shared only with the OpenAI API.

**Comment by enricoros** [comment]
How if we used some local storage of the browser (or similar, like IndexedDB) to store the conversations, on a per-browser basis?

**Comment by bytesuji** [comment]
Yes, was thinking localStorage would be an OK way to do this, you could probably fit a lot of chat history in there if you compressed it. You could also set an expiry on the messages so the oldest ones are deleted to free up space.

Another idea, along the same lines, is to clear the context of old messages once the context window is full, i.e. you hit 8k tokens and start getting `InvalidRequestError`

**Comment by enricoros** [comment]
Closed the original issue (3.5-Turbo) thanks to @koganei. Moved other issues out
